### PR TITLE
Remove large group (#1115)

### PR DESCRIPTION
**Pull request by theofidry**
As far as I can see this group is not actually used and this test is run as part of `e2e`. @maks-rafalko can you confirm?

### CODE REVIEW

**Comment by maks-rafalko** [comment]
in `master`:

`vendor/bin/phpunit --group=e2e` does *not**execute `test_it_runs_on_itself` test

So this PR is quite dangerous, this very large test will be executed with other `e2e` tests while previously it wasn't

We should ask @sanmai why it was added. Probably we don't need it?

**Comment by theofidry** [comment]
When do we ever execute it then? I could not find any occurrence anywhere

File: tests/phpunit/Console/E2ETest.php [hunk 1]
```diff
@@ -125,22 +125,13 @@ protected function tearDown(): void
...
-     * @large
-     */
     public function test_it_runs_on_itself(): void
     {
         if (ini_get('memory_limit') === '-1') {
             $this->markTestSkipped(implode("\n", [
                 'Refusing to run Infection on itself with no memory limit set: it is dangerous.',
                 'To run this test with a memory limit set please use:',
-                'php -dmemory_limit=128M vendor/bin/phpunit --group=large',
+                'php -dmemory_limit=128M vendor/bin/phpunit --group=e2e',
```
Comment (sanmai) [inline tests/phpunit/Console/E2ETest.php]: When this was written, the entire test suite wasn't marked as e2e. So it was correct back then.

**Comment by sanmai** [comment]
The build is failing with:

> Cannot open file ""--exclude-group=integration".php".

File: tests/phpunit/Console/E2ETest.php [hunk 2]
```diff
@@ -125,22 +125,13 @@ protected function tearDown(): void
...
     }
 
-    /**
-     * Longest test: runs under about 160-200 sec
-     *
-     * To be run with:
-     *
-     * php -dmemory_limit=128M vendor/bin/phpunit --group=large
-     *
-     * @large
```
Comment (sanmai) [inline tests/phpunit/Console/E2ETest.php]: I'd rather keep this designation. And explanation to its existence.

File: phpunit.xml.dist [hunk 3]
```diff
@@ -26,12 +28,6 @@
         </whitelist>
     </filter>
 
-    <groups>
-        <exclude>
-            <group>large</group>
-        </exclude>
-    </groups>
```
Comment (sanmai) [inline phpunit.xml.dist]: And so is best to ignore this too, IMO.

**Comment by sanmai** [comment]
About removing, this is a fine test, for example, to run when you want to find dead code.

As it is only few lines of code, I propose to keep it.

**Comment by theofidry** [comment]
Closing in favour of #1120

**Pull request closed without merging**